apis/hive/v1/generics.go

```diff
@@ -0,0 +1,9 @@
+package v1
```

Comment:
nit: Something about this file name is making me leery, but I can't put my finger on it (naming a file after a language feature?) and don't have a better suggestion.

Reply:
Maybe `constraints.go`?

Reply:
I was thinking `conditions.go`, except that I imagine at some point we'll have generic interfacey things for other pieces. Then again, they could be in separate files when we do.